src/wrapper_array.jl

```diff
@@ -82,6 +82,14 @@ function named_size(a::AbstractArray{T,N}) where {T,N}
     L = dimnames(a)
     return NamedTuple{L,NTuple{N,Int}}(size(a))
 end
+function Base.similar(a::NamedDimsArray)
+    dnames = dimnames(a)
+    return NamedDimsArray(similar(parent(a)), dnames)
+end
+function Base.similar(a::NamedDimsArray{L,T}, eltype::Type=T) where {L,T}
```

Comment:
I don't think this is causing failures, but no need for default here -- that's the method just above.
```suggestion
function Base.similar(a::NamedDimsArray, eltype::Type)
```

Reply:
I think we need to remove the defaults from the next method as well, otherwise in both cases it will dispatch to the third one and compute the defaults. I'm working on a fix locally, I'll push it when I get it working + tested on 1.3 as well

Reply:
There's one more quirk here, that I think most of these methods are failing to specialise on the type:
```
julia> nda = NamedDimsArray(rand(10, 100), (:x, :y));

julia> @btime similar($nda, Int);
  257.952 ns (1 allocation: 7.94 KiB)

julia> @btime similar(parent($nda), Int);
  55.287 ns (1 allocation: 7.94 KiB)

julia> Base.similar(a::NamedDimsArray{L,T}, eltype::Type=T) where {L,T} = NamedDimsArray{L}(similar(parent(a), eltype))

julia> @btime similar($nda, Int);
  265.000 ns (1 allocation: 7.94 KiB)

julia> Base.similar(a::NamedDimsArray{L,T}, ::Type{eltype}=T) where {L,T,eltype} = NamedDimsArray{L}(similar(parent(a), eltype))

julia> @btime similar($nda, Int);
  50.553 ns (1 allocation: 7.94 KiB)
```
Not your fault, but it might be worth changing all to explicitly specialise?

Reply:
I pushed a fix changing the second argument to `::Type{eltype}`. Interestingly, I'm not able to reproduce this on my machine on 1.6.1
```julia
julia> nda = NamedDimsArray(rand(10, 100), (:x, :y));                                                                      

julia> @btime similar($nda, Int);                                                                                          
  168.334 ns (1 allocation: 7.94 KiB)

julia> @btime similar(parent($nda), Int);                                                                                  
  157.707 ns (1 allocation: 7.94 KiB)

julia> Base.similar(a::NamedDimsArray{L,T}, eltype::Type=T) where {L,T} = NamedDimsArray{L}(similar(parent(a), eltype))    

julia> @btime similar($nda, Int);                                                                                          
  158.322 ns (1 allocation: 7.94 KiB)

julia> Base.similar(a::NamedDimsArray{L,T}, ::Type{eltype}=T) where {L,T,eltype} = NamedDimsArray{L}(similar(parent(a), eltype))                                                                                                                      

julia> @btime similar($nda, Int);                                                                                          
  160.185 ns (1 allocation: 7.94 KiB)
```

Reply:
Now that I try again, I can't get anything that dramatic, maybe 10% on one machine. I've forgotten which version the initial check was on, maybe that matters? Anyway, thanks for fixing it all the same!